src/api/control/async/asyncMotionProfileController.cpp

```diff
@@ -359,4 +361,116 @@ void AsyncMotionProfileController::startThread() {
 CrossplatformThread *AsyncMotionProfileController::getThread() const {
   return task;
 }
+
+void AsyncMotionProfileController::storePath(std::string idirectory, std::string ipathId) {
+  std::string leftFilePath = makeFilePath(idirectory, ipathId + ".left.csv");
+  std::string rightFilePath = makeFilePath(idirectory, ipathId + ".right.csv");
+  FILE *leftPathFile = fopen(leftFilePath.c_str(), "w");
+  FILE *rightPathFile = fopen(rightFilePath.c_str(), "w");
+
+  // Make sure we can open the file successfully
+  if (leftPathFile == NULL) {
+    LOG_WARN("AsyncMotionProfileController: Couldn't open file " + leftFilePath + " for writing");
+    return;
+  }
+  if (rightPathFile == NULL) {
+    LOG_WARN("AsyncMotionProfileController: Couldn't open file " + rightFilePath + " for writing");
+    return; // cppcheck thinks this is a resource leak, but fclose(NULL) is undefined behavior
```

Comment:
And cppcheck is right because `leftPathFile` is not null.

Reply:
Good catch, fixed